src/components/spaces/show/index.js

```diff
@@ -121,6 +124,19 @@ export default class SpacesShow extends Component {
     this.props.dispatch(spaceActions.destroy(this.props.denormalizedSpace))
   }
 
+  onImportSlurp(slurpObj) {
+    const space = this.props.denormalizedSpace
+
+    const spaceUpdates = parseSlurp(slurpObj, space)
+    let noGraphUpdates = {}
+    if (!space.name) {noGraphUpdates.name = spaceUpdates.name}
+    if (!space.description) {noGraphUpdates.description = spaceUpdates.description}
+    this.props.dispatch(spaceActions.update(this._id(), noGraphUpdates))
+    this.props.dispatch({type: 'ADD_METRICS', items: spaceUpdates.newMetrics, newGuesstimates: spaceUpdates.newGuesstimates})
+    this.props.dispatch(spaceActions.updateGraph(this._id()))
+    this.props.dispatch(simulationActions.runSimulations(this._id(), spaceUpdates.newMetrics))
```

Comment:
This to me feels like it should all be handled from one action, which is the ImportSlurp action.  I realize that doing so would result in one extra render, but it feels more succinct to me.